src/libs/highlevel/README.md

```diff
@@ -0,0 +1,154 @@
+
+`elektraGet` + the type of the value you want to read + `ArrayElement`.
+
+As an example here is the signature of the function to get a string value from an array:
+
+```c
+const char * elektraGetStringArrayElement (Elektra * elektra, const char * name, size_t index);
+```
+
+To get the size of the array you would like to access you can use this function:
+
+```c
+size_t elektraArraySize (Elektra * elektra, const char * name);
+```
+
+For some background information on arrays in Elektra see the [Application Integration](doc/tutorials/application-integration.md) document.
+
+Notice that both the getters for primitive types and the getters for array types do not accept error parameters. The library relies on that you are running a correct Elektra setup. If the configuration is well specified, no runtime errors can occur when reading a value. Therefore the getters do not accept an error variable as argument. If there is however a severe internal error, or you try to access a key which you have not specified correctly, then the library will call `exit(EXIT_FAILURE)` to prevent data inconsistencies or exceptions further down in your application.
+
+You can find the complete list of the available functions for all supported value types in [elektra.h](libelektra/libs/highlevel/elektra.h)
+
+### Writing values to the KDB
+
+Sometimes, after having read a value from the KDB, you will want to write back a modified value. As descibed in [Read values from the KDB](#read-values-from-the-kdb) we follow a naming scheme for getters. The High-Level API provides setters folling an analogous naming scheme as well. For example, to write back a modified string, you can call this function:
+
+```c
+void elektraSetString (Elektra * elektra, const char * name, const char * value);
+```
+
+The counterpart for array-gettes again follows the same naming scheme:
+
+```c
+void elektraSetStringArrayElement (Elektra * elektra, const char * name, const char * value, size_t index);
+```
+
+Be sure not to access indexes outside of the arrays bounds. The same rules a described in [Read values from the KDB](#read-values-from-the-kdb) apply here, meaning, that you are responsible for providing a complete and correct specification (see [Application Integration](doc/tutorials/application-integration.md)). If you try to access a key that you have not specified, the library will call `exit(EXIT_FAILURE)`.
+
+## Example
+```c
+#include <stdio.h>
+#include "elektra.h"
+
+int main(int argc, char ** argv)
+{
+  ElektraError * error = NULL;
+  Elektra * elektra = elektraOpen ("/sw/yourApplication", &error);
+
+  if (error != NULL) 
+  {
+    printf ("Sorry, there seems to be an error with your Elektra setup: %s\n", elektraErrorDescription (error));
+    printf ("Will exit now...\n");
+    exit (EXIT_FAILURE);
+    elektraErrorFree (error);
```

Comment:
An error occured and the handle is NULL so elektraClose is not necessary. elektraErrorFree really makes no sense here, swapped the two lines.